docs/dependent-apps.md

```diff
@@ -0,0 +1,289 @@
+# Using the Resin Supervisor for managing dependent applications
+
+Since version 2.5.0 the Resin Supervisor can act as a proxy for dependent apps.
+
+Only Supervisors after version 2.5.0 have this functionality, and some of the endpoints appeared in later versions (we've noted it down where this is the case).
+
+## What is a dependent application
+
+A **dependent application** is a resin application that targets devices not capable of interacting directly with the Resin API - the reasons can be several, the most common are:
+
+- no direct Internet capabilities
+- not able to run Resin OS (being a microcontroller, for example)
+
+The **dependent application** is scoped under a resin application, which gets the definition of **gateway application**.
+
+The **gateway application** is responsible for detecting, provisioning and managing **dependent devices** belonging to one of its **dependent applications**. This is possible leveraging a new set of endpoints exposed by the Resin Supervisor.
+
+A dependent application follows the same development cycle of a conventional resin application:
```

Comment:
It's also mentioned below that a dependent application:

> consists in a Docker application
> the actual firmware/business logic must be stored in the `/assets` folder within the built docker image.

So if I understood this correctly, the resulting dependent application Docker image is only used to build, package and deliver the firmware in the `/assets` folder on the device, is that correct? If so, we might want to express this more clearly in the doc and stress that, in contrast with both conventional and gateway applications, a `git push` for a dependent app will not result in a new running container on the device.